frontend/server/src/Controllers/Course.php

```diff
@@ -2058,7 +2058,18 @@ public static function apiStudentProgress(\OmegaUp\Request $r): array {
                 'assignmentNotFound'
             );
         }
+        return [
+            'problems' => self::getProblemsBySelectedAssignment(
+                $assignment,
+                $resolvedIdentity
+            ),
+        ];
+    }
 
+    private static function getProblemsBySelectedAssignment(
+        $assignment,
```

Comment:
Acá se queja psalm de que no estás declarando el tipo del parámetro, hayq que agregarlo:
```suggestion
        \OmegaUp\DAO\VO\Assignments $assignment,
```